src/Engine.php

```diff
@@ -161,6 +161,23 @@ public function search(string $query, string $indexName, array $searchParams): a
         return $this->client->index($indexName)->rawSearch($query, $searchParams);
     }
 
+    /**
+     * Search the index and returns the document identifier.
+     *
+     * @param string                          $query
+     * @param string                          $indexName
+     * @param array<string, int|string|array> $searchParams
+     *
+     * @return array<string, int|string|array>
+     */
+    public function searchIds(string $query, string $indexName, array $searchParams)
+    {
+        $primaryKey = $this->client->index($indexName)->fetchPrimaryKey();
+        $result = $this->search($query, $indexName, $searchParams);
```

Comment:
As I'm a fresh MS user: Would it improve performance or something to limit the search to only fetch the $primaryKey attribute?

https://docs.meilisearch.com/reference/features/search_parameters.html#attributes-to-retrieve

Something like this:

~~~php
if (!isset($searchParams['attributesToRetrieve'])) {
    $searchParams['attributesToRetrieve'] = [$primaryKey];
}
~~~

Reply:
This is indeed not optimized! Unfortunately, the primary key is not accessible in the search response so we cannot use your suggestion, except if I missed something. I need to find another way.

Reply:
I've realized it's more complicated than I thought. The `search()` method was badly adapted for MeiliSearch (first coming from [Algolia search bundle](https://github.com/algolia/search-bundle/blob/6f57eed309fc3b9e129c2643570bf64db892501b/src/Services/AlgoliaSearchService.php#L204-L239).
We indeed don't use `objectID` with MeiliSearch but your primary key (so the unique field of your documents) is inferred because containing `id`, so often, the primary key is the simple `id` field.

But, according to the way of handling the unique field, it means we are not able to know the primary key (= the unique field) during the search. And we clearly need to know it according to our code base!

Two possibilities:
- we are doing 2 HTTP calls during the `search()`: this is a quick fix and your PR (#60) is a temporary workaround for users having performance issues. I don't really like this solution if the second point can be applied.
- removing `objectID` from our code base, I mean, in terms of usage. Related to #58. It looks like being a big task to do and I don't have the time to do it, unfortunately, but this would be a good step in this package. I'm opening an issue.